File: server-csharp/Map/Pieces/Platform.cs
using System.Collections.Generic;
using System.Numerics;
using SpacetimeDB;

public static partial class Module
{
    public static readonly List<DbVector3> PlatformConvexHull0Vertices = new List<DbVector3>
    {
        // Bottom face (Y = 2.15 - 0.25 = 1.9)
        new DbVector3(-2.375f, 1.9f, 11.875f), // Bottom-Far-Left
        new DbVector3( 2.375f, 1.9f, 11.875f), // Bottom-Far-Right
        new DbVector3( 2.375f, 1.9f,  8.125f), // Bottom-Near-Right
        new DbVector3(-2.375f, 1.9f,  8.125f), // Bottom-Near-Left

        // Top face (Y = 2.15 + 0.25 = 2.4)
        new DbVector3(-2.375f, 2.4f, 11.875f), // Top-Far-Left
        new DbVector3( 2.375f, 2.4f, 11.875f), // Top-Far-Right
        new DbVector3( 2.375f, 2.4f,  8.125f), // Top-Near-Right
        new DbVector3(-2.375f, 2.4f,  8.125f), // Top-Near-Left
    };

    public static readonly ConvexHullCollider PlatformConvexHull0 = new ConvexHullCollider
    {
        VerticesLocal = PlatformConvexHull0Vertices
    };

    public static readonly List<ConvexHullCollider> PlatformConvexHulls = new List<ConvexHullCollider>
    {
        PlatformConvexHull0,
    };

    public static readonly ComplexCollider PlatformCollider = new ComplexCollider
    {
        ConvexHulls = PlatformConvexHulls,
        CenterPoint = new DbVector3(0.0f, 2.15f, 10.0f)
    };

}

File: server-rust/spacetimedb/src/Maps/Pieces/Platform.rs
use crate::*;

pub static PlatformConvexHull0Vertices: &[DbVector3] = &[
    DbVector3 { x: -2.375, y: 1.9, z: 11.875 }, DbVector3 { x:  2.375, y: 1.9, z: 11.875 }, DbVector3 { x:  2.375, y: 1.9, z:  8.125 }, DbVector3 { x: -2.375, y: 1.9, z:  8.125 },
    DbVector3 { x: -2.375, y: 2.4, z: 11.875 }, DbVector3 { x:  2.375, y: 2.4, z: 11.875 }, DbVector3 { x:  2.375, y: 2.4, z:  8.125 }, DbVector3 { x: -2.375, y: 2.4, z:  8.125 },
];

pub static PlatformConvexHull0TriangleIndicesLocal: &[i32] = &[0, 2, 1, 1, 2, 6, 0, 1, 4, 6, 2, 3, 2, 0, 3, 0, 4, 3, 1, 6, 5, 6, 4, 5, 4, 1, 5, 4, 6, 7, 6, 3, 7, 3, 4, 7];

pub fn PlatformCollider() -> ComplexCollider {
    let platform_convex_hull_0: ConvexHullCollider = ConvexHullCollider { vertices_local: PlatformConvexHull0Vertices.to_vec(), triangle_indices_local: PlatformConvexHull0TriangleIndicesLocal.to_vec(), margin: 0.0 };
    let platform_convex_hulls: Vec<ConvexHullCollider> = vec![platform_convex_hull_0];
    ComplexCollider { convex_hulls: platform_convex_hulls, center_point: DbVector3 { x: 0.0, y: 2.15, z: 10.0 } }
}
